### PR TITLE
Address suggestions from review

### LINKED ISSUE #2: How can SunPy support and encourage more solar observing instruments to provide affiliated packages?
### Provide a general description of the issue or problem.

We need to address this

### LINKED ISSUE #3: how to control and ensure the quality of SunPy as the number of affiliated packages increases? Is it possible to have a discussion? This would be a better guide for the development teams.
### Provide a general description of the issue or problem.

We need to address this

### LINKED ISSUE #4: Expressing explicitly how the interoperability works, e.g., standard data format and same interfaces, etc
### Provide a general description of the issue or problem.

We need to address this

### LINKED ISSUE #5: Explain how you achieve reproducibility.
### Provide a general description of the issue or problem.

We need to address this

### LINKED ISSUE #6: Create a notebook version of the AF package script?
### Provide a general description of the issue or problem.

Not sure if its useful.

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #4
Fixes #5
Fixes #6 
Fixes #3
Fixes #2 

This PR addresses all of the suggestions from the referee.

When making changes to the paper, please use the `changes` package to make additions/deletions/comment: https://mirror2.sandyriver.net/pub/ctan/macros/latex/contrib/changes/changes.english.pdf

### CODE REVIEW

File: main.tex [hunk 1]
```diff
@@ -67,6 +67,10 @@
 \def\corrAuthor{Barnes, Will T.}
 \def\corrEmail{will.t.barnes@nasa.gov}
 
+% This is for tracking changes
+\definechangesauthor[name={Will Barnes}, color=orange]{WTB}
+\definechangesauthor[name={Nabil Freij}, color=orange]{NF}
```
Comment (nabobalis) [inline main.tex:72]:
```suggestion
\definechangesauthor[name={Nabil Freij}, color=blue]{NF}
```
:P

File: sections/1-introduction.tex [hunk 2]
```diff
@@ -2,11 +2,11 @@ \section{Introduction}
...
+The vision of the project is to build a diverse and inclusive solar physics and heliophysics community that supports scientific discovery and \added[id=WTB]{enables} reproducibility through the development of accessible, open-source software \citep{bobra_monica_2020_7020094}.
 To achieve this mission and to make this vision a reality, the \sunpyproj maintains and guides the development of a number of Python packages including the \sunpypkg core package, and organizes educational activities around the use of Python for solar-physics research.
 
 As the scientific Python environment matured in the early 2010s \citep{Hunter:2007, harris2020array, 2020SciPy-NMeth}, the development of a Python package devoted to solar physics became viable.
-This led to the founding of the \sunpyproj led by scientists at NASA Goddard Space Flight Center in 2014.
+This led to the founding of the \sunpyproj led by scientists at NASA Goddard Space Flight Center in 2011.
```
Comment (nabobalis) [inline sections/1-introduction.tex:9]:
```suggestion
This led to the founding of the \sunpyproj led by scientists at NASA Goddard Space Flight Center in \added[id=WTB]{2011}.
```

File: sections/2-code.tex [hunk 3]
```diff
@@ -7,6 +7,7 @@ \subsection{The \sunpypkg core package}
 The \sunpypkg package is the central pillar of the \sunpyproj \citep{sunpy_community2020} and is designed to provide the fundamental tools for accessing, loading, and interacting with solar physics data in Python.
 The primary components of the \sunpypkg package are summarized graphically in \autoref{fig:sunpy-core-summary} and described briefly in the following paragraphs.
 For a more in-depth description of each of these components, see \citet[Section 4]{sunpy_community2020}.
+\added[id=NF]{Additionally, the full description of the \sunpypkg Application Programming Interface (API) is provided in the hosted online documentation}\footnote{https://docs.sunpy.org/en/stable/}.
```
Comment (nabobalis) [inline sections/2-code.tex:10]:
```suggestion
\added[id=NF]{If one desires the Application Programming Interface of \sunpypkg, this is provided in the documentation}\footnote{https://docs.sunpy.org/en/stable/}.
```

Shorter at least?

File: sections/2-code.tex [hunk 4]
```diff
@@ -15,21 +16,47 @@ \subsection{The \sunpypkg core package}
...
+\added[id=WTB]{Section 4.1.1 of} \citep{sunpy_community2020} \added[id=WTB]{also provides a more detailed discussion of the data sources that \Fido searches by default.}
+\added[id=WTB]{Additionally, \Fido can also be extended to search additional data sources that may not be included in \sunpypkg (e.g. the Solar Orbiter Archive, see} \autoref{sssec:current-ecosystem}).
 As illustrated in the leftmost column of \autoref{fig:sunpy-core-summary}, attributes such as time, wavelength, and instrument name, among others, can be used to filter these search results.
 By providing a single interface to many disparate data sources, \sunpypkg, via \Fido, easily enables multi-instrument research workflows.
 
-Once a user has downloaded data, the \code{TimeSeries} and \code{Map} data objects can be used to load time series and two-dimensional coordinate aware image data, respectively.
+Once a user has downloaded data, the \code{TimeSeries} and \code{Map} data objects can be used to load time series and two-dimensional image data, respectively.
```
Comment (nabobalis) [inline sections/2-code.tex:29]: Are not all the 2D image data, coordinate aware?

File: sections/2-code.tex [hunk 5]
```diff
@@ -15,21 +16,47 @@ \subsection{The \sunpypkg core package}
...
+
+\subsubsection{Release Schedule}
+\label{sssection:releases}
+
 There is a new release of the core package with feature enhancements approximately every six months.
 Every other release is designated a long term support (LTS) release and receives bug fixes for a year rather than for six months.
 Additionally, there are bug fix releases every month.
+\added[id=WTB]{For each release a digital object identifier (DOI) is automatically generated and a record is created on Zenodo.}\footnote{The most current release on Zenodo can be found here: \url{https://doi.org/10.5281/zenodo.7314636}}
+\added[id=WTB]{By providing regularly scheduled-scheduled, versioned releases of \sunpypkg, the \sunpyproj enables reproducibility.
+For example, if a researcher is attempting to reproduce a result from a paper that used sunpy v2.0.2, she can create a new virtual environment and install that exact version of sunpy, even if the current version is many versions ahead of v2.0.2.}
```
Comment (nabobalis) [inline sections/2-code.tex:54]:
```suggestion
For example, if a researcher is attempting to reproduce a result from a paper that used \sunpypkg v2.0.2, she can create a new virtual environment and install that exact version of \sunpypkg, even if the current version is many versions ahead of v2.0.2.}
```
Maybe?

File: sections/4-community.tex [hunk 6]
```diff
@@ -26,10 +26,10 @@ \subsubsection{Communication Channels}
 These include:
 
 \begin{itemize}
-    \item Multiple GitHub repositories for bug reports and feature requests. These are listed under the sunpy GitHub organization: \url{https://github.com/sunpy}/.
-    \item Real time messaging on Matrix at \url{https://matrix.to/#/#sunpy:openastronomy.org}.
-    \item Mailing Lists at \url{https://groups.google.com/g/sunpy}
-    \item An online community forum at \url{https://community.openastronomy.org/}
+    \item Multiple GitHub repositories for bug reports and feature requests. These are listed under the \sunpyproj GitHub organization: \footnote{\url{https://github.com/sunpy}}.
```
Comment (nabobalis) [inline sections/4-community.tex:29]:
```suggestion
    \item Multiple GitHub repositories for bug reports and feature requests. These are listed under the \sunpyproj GitHub organization \footnote{\url{https://github.com/sunpy}}.
```

File: sections/4-community.tex [hunk 7]
```diff
@@ -26,10 +26,10 @@ \subsubsection{Communication Channels}
...
+    \item Real time messaging\footnote{Links to join the Matrix chat can be found at \url{https://sunpy.org/help.html}}.
+    \item Mailing lists\footnote{\url{https://groups.google.com/g/sunpy}}
+    \item An online community forum at \footnote{\url{https://community.openastronomy.org/}}
```
Comment (nabobalis) [inline sections/4-community.tex:32]:
```suggestion
    \item An online community forum. \footnote{\url{https://community.openastronomy.org/}}
```

File: sections/2-code.tex (hunk 3 above)
Comment (wtbarnes) [reply sections/2-code.tex:10]: I am not a fan of that wording

Comment (nabobalis) [reply sections/2-code.tex:10]: Hmm, ok. Let me try again.

File: sections/2-code.tex [hunk 8]
```diff
@@ -7,6 +7,7 @@ \subsection{The \sunpypkg core package}
 The \sunpypkg package is the central pillar of the \sunpyproj \citep{sunpy_community2020} and is designed to provide the fundamental tools for accessing, loading, and interacting with solar physics data in Python.
 The primary components of the \sunpypkg package are summarized graphically in \autoref{fig:sunpy-core-summary} and described briefly in the following paragraphs.
 For a more in-depth description of each of these components, see \citet[Section 4]{sunpy_community2020}.
+\added[id=NF]{information on the Application Programming Interface for \sunpypkg is provided by the documentation}\footnote{https://docs.sunpy.org/en/stable/}.
```
Comment (nabobalis) [inline sections/2-code.tex:10]: How do we feel about this?

File: sections/2-code.tex [hunk 9]
```diff
@@ -15,21 +16,47 @@ \subsection{The \sunpypkg core package}
     \label{fig:sunpy-core-summary}
 \end{figure}
 
+\subsubsection{Components of the Core Package}
+\label{sssec:components-of-the-core-package}
+
 To search for and download data, \sunpypkg provides the \Fido interface for searching across a variety of data providers (e.g., the Virtual Solar Observatory (VSO)\footnote{\url{https://sdac.virtualsolar.org/cgi/search}}, or the Joint Science Operations Center (JSOC)\footnote{\url{http://jsoc.stanford.edu}}) maintained within the solar community.
+\added[id=WTB]{A complete list of all supported data sources is provided in the documentation for using \Fido}\footnote{\url{https://docs.sunpy.org/en/stable/guide/acquiring_data/fido.html}}.
+\added[id=WTB]{Section 4.1.1 of} \citep{sunpy_community2020} \added[id=WTB]{also provides a more detailed discussion of the data sources that \Fido searches by default.}
```
Comment (nabobalis) [inline sections/2-code.tex:24]:
```suggestion
\added[id=WTB]{Whereas, Section 4.1.1 of} \citep{sunpy_community2020} \added[id=WTB]{also provides a more detailed discussion of the data sources that \Fido searches by default.}
```

File: sections/2-code.tex [hunk 10]
```diff
@@ -15,21 +16,47 @@ \subsection{The \sunpypkg core package}
...
 Each \code{Map} object instance also carries with it the corresponding coordinate frame of that image and the coordinate of the observer as defined by the position of the observatory given in the associated metadata.
 
+\subsubsection{Testing Infrastructure}
+\label{sssec:testing-infrastructure}
+
+\added[id=WTB]{\sunpypkg includes thousands of unit and integration tests that are run using the \code{pytest} testing framework as well as \code{tox} for ensuring tests are run in isolated environments.
+This test suite also includes tests of the documentation.
+This test suite is run on every pull request opened on the \sunpypkg \github using \github Actions to ensure that contributions to the codebase do not lead to unexpected regressions.
+Additionally, we also make use of the Circle CI platform.
+A full description of our testing practices can be found in our developer documentation.}\footnote{A complete guide to running the tests and the associated infrastructure can be found here:\url{https://docs.sunpy.org/en/latest/dev_guide/contents/tests.html}}
```
Comment (nabobalis) [inline sections/2-code.tex:44]:
```suggestion
\added[id=WTB]{\sunpypkg includes thousands of unit and integration tests that are run using the \code{pytest} testing framework and includes tests of the documentation.
This test suite is run on every pull request opened on the \sunpypkg \github using \github Actions to ensure that contributions to the codebase do not lead to unexpected regressions.
A full description of our testing practices can be found in our developer documentation.}\footnote{A complete guide to running the tests and the associated infrastructure can be found here:\url{https://docs.sunpy.org/en/latest/dev_guide/contents/tests.html}}
```

File: sections/2-code.tex [hunk 11]
```diff
@@ -15,21 +16,47 @@ \subsection{The \sunpypkg core package}
...
+
+\subsubsection{Release Schedule}
+\label{sssec:release-schedule}
+
 There is a new release of the core package with feature enhancements approximately every six months.
 Every other release is designated a long term support (LTS) release and receives bug fixes for a year rather than for six months.
 Additionally, there are bug fix releases every month.
+\added[id=WTB]{For each release a digital object identifier (DOI) is automatically generated and a record is created on Zenodo.}\footnote{The most current release on Zenodo can be found here: \url{https://doi.org/10.5281/zenodo.7314636}}
+\added[id=WTB]{By providing regularly scheduled-scheduled, versioned releases of \sunpypkg, the \sunpyproj enables reproducibility.
```
Comment (nabobalis) [inline sections/2-code.tex:53]:
```suggestion
\added[id=WTB]{By providing regularly scheduled, versioned releases of \sunpypkg, the \sunpyproj enables reproducibility.
```

File: sections/2-code.tex [hunk 12]
```diff
@@ -15,21 +16,47 @@ \subsection{The \sunpypkg core package}
...
+For example, if a researcher is attempting to reproduce a result from a paper that used \sunpypkg v2.0.2, she can create a new virtual environment and install that exact version of \sunpypkg, even if the current version is many versions ahead of v2.0.2.}
+
+\added[id=WTB]{This release process is completely automated through GitHub Actions.}\footnote{The \github Actions templates used by \sunpypkg and the affiliated packages within the \sunpyproj are available here: \url{https://github.com/OpenAstronomy/github-actions-workflows}}
```
Comment (nabobalis) [inline sections/2-code.tex:56]:
```suggestion
\added[id=WTB]{This release process is completely automated through GitHub Actions.}\footnote{The \github Actions templates used are available here: \url{https://github.com/OpenAstronomy/github-actions-workflows}}
```

File: sections/2-code.tex [hunk 13]
```diff
@@ -15,21 +16,47 @@ \subsection{The \sunpypkg core package}
...
+\added[id=WTB]{When a release is tagged, an action is triggered that tests the package on all supported versions of Python and all supported operating systems.
+If the packages build successfully, they are automatically uploaded to to Python Package Index (PyPi).
+Subsequently, a PR is automatically opened on the \code{conda-forge} feedstock to upload the release to \code{conda-forge}.}
```
Comment (nabobalis) [inline sections/2-code.tex:59]:
```suggestion
```

File: sections/2-code.tex [hunk 14]
```diff
@@ -15,21 +16,47 @@ \subsection{The \sunpypkg core package}
...
+\added[id=WTB]{For each release a digital object identifier (DOI) is automatically generated and a record is created on Zenodo.}\footnote{The most current release on Zenodo can be found here: \url{https://doi.org/10.5281/zenodo.7314636}}
+\added[id=WTB]{By providing regularly scheduled-scheduled, versioned releases of \sunpypkg, the \sunpyproj enables reproducibility.
+For example, if a researcher is attempting to reproduce a result from a paper that used \sunpypkg v2.0.2, she can create a new virtual environment and install that exact version of \sunpypkg, even if the current version is many versions ahead of v2.0.2.}
+
+\added[id=WTB]{This release process is completely automated through GitHub Actions.}\footnote{The \github Actions templates used by \sunpypkg and the affiliated packages within the \sunpyproj are available here: \url{https://github.com/OpenAstronomy/github-actions-workflows}}
+\added[id=WTB]{When a release is tagged, an action is triggered that tests the package on all supported versions of Python and all supported operating systems.
+If the packages build successfully, they are automatically uploaded to to Python Package Index (PyPi).
```
Comment (nabobalis) [inline sections/2-code.tex:58]:
```suggestion
If the packages build successfully, they are automatically uploaded to to Python Package Index (PyPi) and subsequently the release is updated on \code{conda-forge}.}
```

File: sections/2-code.tex [hunk 15]
```diff
@@ -79,6 +106,8 @@ \subsubsection{Application Process}
 
 In all cases, the goal of the affiliated package review process is to broaden the ecosystem of tools for solar data analysis in Python.
 These criteria are not meant to be exclusionary, but rather to ensure interoperability and consistency across the ecosystem for the benefit of both users and developers.
+\added[id=NF]{Interoperability, in this context, means that affiliated packages should make use of the existing \sunpypkg core data structures, (e.g., \code{Map} and \code{Timeseries}), wherever possible in lieu of their own custom data structures.
```
Comment (nabobalis) [inline sections/2-code.tex:109]:
```suggestion
\added[id=NF]{Interoperability, in this context, means that affiliated packages should make use of the existing \sunpypkg core data structures, (e.g., \code{Map} and \code{Timeseries}), in lieu of their own custom data structures.
```

File: sections/2-code.tex (hunk 10 above)
Comment (nabobalis) [reply sections/2-code.tex:44]: I wanted to cut down some of the info.

**Comment by nabobalis** [comment]
The last commit I added, needs major improvements.

File: sections/2-code.tex (hunk 8 above)
Comment (nabobalis) [inline sections/2-code.tex:10]:
```suggestion
\added[id=NF]{The \sunpypkg provides detailed Application Programming Interface (API) documentation}\footnote{https://docs.sunpy.org/en/stable/}.
```

File: sections/2-code.tex [hunk 16]
```diff
@@ -15,21 +16,44 @@ \subsection{The \sunpypkg core package}
...
+\label{sssec:release-schedule}
+
 There is a new release of the core package with feature enhancements approximately every six months.
 Every other release is designated a long term support (LTS) release and receives bug fixes for a year rather than for six months.
 Additionally, there are bug fix releases every month.
+\added[id=WTB]{For each release a digital object identifier (DOI) is automatically generated and a record is created on Zenodo.}\footnote{The most current release on Zenodo can be found here: \url{https://doi.org/10.5281/zenodo.7314636}}
+\added[id=WTB]{By providing regularly scheduled, versioned releases of \sunpypkg, the \sunpyproj enables reproducibility.
+For example, if a researcher is attempting to reproduce a result from a paper that used \sunpypkg v2.0.2, she can create a new virtual environment and install that exact version of \sunpypkg, even if the current version is many versions ahead of v2.0.2.}
+
+\added[id=WTB]{This release process is completely automated through GitHub Actions.}\footnote{The \github Actions templates used are available here: \url{https://github.com/OpenAstronomy/github-actions-workflows}}
```
Comment (nabobalis) [inline sections/2-code.tex:54]:
```suggestion
\added[id=WTB]{This release process is completely automated through \github Actions.}\footnote{The \github Actions templates used are available here: \url{https://github.com/OpenAstronomy/github-actions-workflows}}
```

File: sections/2-code.tex [hunk 17]
```diff
@@ -50,9 +74,14 @@ \subsection{Affiliated Packages}
...
 While these packages are not yet affiliated, the \sunpyproj has assisted in coordinating development efforts between these teams in order to foster a more interoperable ecosystem.
 
+\added[id=NF]{As the number of \sunpyproj affiliated packages increases, it is important to consider how to ensure that the quality of the affiliated packages does not change.
+Quality is somewhat subjective when it comes to software, as the Application process below outlines, several metrics enable a cursory glance at the quality of a package.
+In the future (due to the Roses Grant), work will undertaken to automate the collection of these metrics and report them back to the wider community and we would like to start doing annual re-reviews of all affiliated packages.
+There is much work to be done on this front and we welcome any contributions to this effort.}
```
Comment (nabobalis) [inline sections/2-code.tex:83]:
```suggestion
\added[id=NF]{The goal of the \sunpyproj is to slowly increase the number of affiliated packages.
This brings the challenge of ensuring that the quality of all affiliated packages never declines.
Quality is somewhat subjective when it comes to software, as the Application Process below outlines, several metrics enable a cursory glance at the quality of a package.
In the future (funded by a Roses Grant), automation of these metrics and collating more information on the \sunpyproj website is planned.
In addition, we would like to start doing annual re-reviews of all affiliated packages.
There is much work to be done on this front and we welcome any contributions to this effort.}
```

File: sections/2-code.tex [hunk 18]
```diff
@@ -79,6 +110,8 @@ \subsubsection{Application Process}
 
 In all cases, the goal of the affiliated package review process is to broaden the ecosystem of tools for solar data analysis in Python.
 These criteria are not meant to be exclusionary, but rather to ensure interoperability and consistency across the ecosystem for the benefit of both users and developers.
+\added[id=NF]{Interoperability, in this context, means that affiliated packages should make use of the existing \sunpypkg core data structures, (e.g., \code{Map} and \code{Timeseries}), in lieu of their own custom data structures.
```
Comment (nabobalis) [inline sections/2-code.tex:113]:
```suggestion
\added[id=NF]{Interoperability in this context, means that affiliated packages should make use of the existing \sunpypkg core data structures, (e.g., \code{Map} and \code{Timeseries}), in lieu of their own custom data structures.
```

File: sections/2-code.tex [hunk 19]
```diff
@@ -96,6 +129,8 @@ \subsubsection{Current Ecosystem}
 \end{itemize}
 
 To demonstrate how several of the affiliated packages can be used together with \sunpypkg in a scientific workflow, we show an example in \autoref{fig:affiliated-package-showcase} of how coronal loop structures can be analyzed using potential magnetic field extrapolations and multi-point extreme ultraviolet (EUV) observations.
+\added[id=WTB]{We have included a Jupyter notebook that illustrates each step of this workflow in the GitHub repository that accompanies this paper.}\footnote{The GitHub repository for this paper, including the complete text and all code to generate Figure 2, can be found at \url{https://github.com/sunpy/sunpy-frontiers-paper}}
```
Comment (nabobalis) [inline sections/2-code.tex:132]:
```suggestion
\added[id=WTB]{We have included a Jupyter notebook that illustrates each step of this workflow in the \github repository that accompanies this paper.}\footnote{The \github repository for this paper, including the complete text and all code to generate Figure 2, can be found at \url{https://github.com/sunpy/sunpy-frontiers-paper}}
```

File: sections/2-code.tex [hunk 20]
```diff
@@ -96,6 +129,8 @@ \subsubsection{Current Ecosystem}
 \end{itemize}
 
 To demonstrate how several of the affiliated packages can be used together with \sunpypkg in a scientific workflow, we show an example in \autoref{fig:affiliated-package-showcase} of how coronal loop structures can be analyzed using potential magnetic field extrapolations and multi-point extreme ultraviolet (EUV) observations.
+\added[id=WTB]{We have included a Jupyter notebook that illustrates each step of this workflow in the \github repository that accompanies this paper.}\footnote{The \github repository for this paper, including the complete text and all code to generate Figure 2, can be found at \url{https://github.com/sunpy/sunpy-frontiers-paper}}
```
Comment (nabobalis) [inline sections/2-code.tex:132]:
```suggestion
\added[id=WTB]{We have included a Jupyter notebook that illustrates each step of this workflow in the \github repository that accompanies this paper.}\footnote{The \github repository for this paper, including the complete text and all code to generate Figure \autoref{fig:affiliated-package-showcase}, can be found at \url{https://github.com/sunpy/sunpy-frontiers-paper}}
```
I wonder if this will break the build.

File: sections/6-conclusion.tex [hunk 21]
```diff
@@ -5,3 +5,7 @@ \section{Conclusion}
 In particular, we have discussed how the \sunpypkg package and the wider set of affiliated packages form an ecosystem for performing solar physics research in Python and the types of analysis that such an ecosystem enables (see \autoref{fig:affiliated-package-showcase}).
 Finally, we have summarized a tentative roadmap to steer the direction of the \sunpyproj in the coming years.
 Importantly, we hope that such a high level description will provide a more clear understanding of the \sunpyproj and the wider ecosystem and will encourage contributions of all forms from the global solar physics community.
+
+\added[id=NF]{The authors of this paper want to end on an explicit call to instrument teams to consider developing affiliated packages.
+The \sunpyproj provides a number of benefits to instrument teams, starting off with a community of developers and users who will provide support and feedback on the development of your package.
+In addition, a Python package template is provided to help get you started and in future will include features such as \textbf{X} and \textbf{Y} (funded by the ROSES Grant).}
```
Comment (nabobalis) [inline sections/6-conclusion.tex:11]:
```suggestion
\added[id=NF]{We want to end on an explicit call to instrument teams to consider developing affiliated packages.
The \sunpyproj provides a number of benefits to instrument teams, starting off with a community of developers and users who will provide feedback on the development of your package.
While we are still primarily a team of volunteers, we can provide (limited) code review, Python or packaging support. 
In addition, a Python package template is being written to simplify the Python packaging process and will enable automatic updates as changes are made.}
```
